h2o-core/src/main/java/water/H2O.java

```diff
@@ -2195,8 +2201,24 @@ public static void main( String[] args ) {
     Log.debug("    Start network services: " + (time10 - time9) + "ms");
     Log.debug("    Cloud up: " + (time11 - time10) + "ms");
     Log.debug("    Start GA: " + (time12 - time11) + "ms");
+
+    if (!ARGS.isEmbeddedApp) {
+      keepH2OAliveThread.start();
```

Comment:
@michalkurka When we ensures that all long-running threads were daemon, H2O actually stopped when it reached end of `main` method. This ensures that we keep h2o running.

Also it ensures that for all existing users of H2O, also the ones who uses H2O as embedded app, will not be affected.

Sparkling Water will set `embedded = true` to ensure we don't do this wait.